From: Amit Shah <amit.shah@redhat.com>
Date: Thu, 29 Mar 2012 12:58:05 +0530
Subject: [PATCH] virtio-pci: switch to PM ops macro to initialise PM functions
Git-commit: f878d0be229ab129bc4b224191f217106a4b0bc1
References: FATE#314135
Patch-Mainline: v3.4

Use the SET_SYSTEM_SLEEP_PM_OPS macro to initialise the suspend/resume
functions in the new PM API.

Signed-off-by: Amit Shah <amit.shah@redhat.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/virtio/virtio_pci.c |    7 +------
 1 files changed, 1 insertions(+), 6 deletions(-)

diff --git a/drivers/virtio/virtio_pci.c b/drivers/virtio/virtio_pci.c
index 0fa4f85..2e03d41 100644
--- a/drivers/virtio/virtio_pci.c
+++ b/drivers/virtio/virtio_pci.c
@@ -768,12 +768,7 @@ static int virtio_pci_restore(struct device *dev)
 }
 
 static const struct dev_pm_ops virtio_pci_pm_ops = {
-	.suspend	= virtio_pci_freeze,
-	.resume		= virtio_pci_restore,
-	.freeze		= virtio_pci_freeze,
-	.thaw		= virtio_pci_restore,
-	.restore	= virtio_pci_restore,
-	.poweroff	= virtio_pci_freeze,
+	SET_SYSTEM_SLEEP_PM_OPS(virtio_pci_freeze, virtio_pci_restore)
 };
 #endif
 
-- 
1.7.4.2

